### PR TITLE
TF import changes

### PR DESCRIPTION
**Pull request by raver119**
This PR adds some changes for TF import:

1) names persistance for vars/nodes
2) libnd4j output mode is VARIABLE_SPACE now by default
3) reverse lookup for String->Integer ids conversion

### CODE REVIEW

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java [hunk 1]
```diff
@@ -4215,7 +4216,29 @@ protected int asFlatNode(String name,@NonNull SameDiff scope, @NonNull FlatBuffe
         return flatNode;
     }
 
-    protected int asFlatNode(@NonNull DifferentialFunction node, @NonNull FlatBufferBuilder bufferBuilder,List<SDVariable> variables) {
+    public static Pair<String, Integer> parseVariable(@NonNull String varName) {
```
Comment (agibsonccc) [inline nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java]: Maybe explain the variable format and expectations?

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/functions/DifferentialFunction.java [hunk 2]
```diff
@@ -64,6 +64,10 @@
     @Getter
     private String instanceId;
 
+    @Getter
+    @Setter
```
Comment (agibsonccc) [inline nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/functions/DifferentialFunction.java]: Did we want to replace instanceId here?

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java (hunk 1 above)
Comment (raver119) [reply nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java]: There's tests for it, it just extracts index part out of string name.

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/functions/DifferentialFunction.java (hunk 2 above)
Comment (raver119) [reply nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/functions/DifferentialFunction.java]: Now I'm not so sure it's a good idea tbh. Having some unique ids somewhere is usually good thing for java world.

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java (hunk 1 above)
Comment (raver119) [reply nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java]: But ye, i'll add javadoc now.

Comment (agibsonccc) [reply nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java]: Javadoc != unit tests :).

**Pull request merged**